.github/workflows/build-docker.yml

```diff
@@ -39,13 +46,26 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-single-buildx
 
+      - if: ${{ github.ref == 'refs/heads/staging' }}
+        name: Build staging image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          builder: ${{ steps.buildx.outputs.name }}
+          file: docker/Dockerfile.staging
+          push: true
+          tags: |
+                  ${{ steps.prep.outputs.tagged_image }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
```

Comment:
Looks like it doesn't differ at all from the step below (with production build).
Except the branch name and Dockerfile name, which is also the same for production and stating.